dev/env.multi-cluster-defaults

```diff
@@ -0,0 +1,14 @@
+export FLEET_E2E_NS=fleet-local
+export FLEET_E2E_NS_DOWNSTREAM=fleet-default
+
+export FLEET_E2E_CLUSTER=k3d-upstream
+export FLEET_E2E_CLUSTER_DOWNSTREAM=k3d-downstream
+
+export GIT_HTTP_USER=fleet-ci
+export GIT_HTTP_PASSWORD=foo
+
+export CI_OCI_USERNAME=fleet-ci
+export CI_OCI_PASSWORD=foo
+export CI_OCI_CERTS_DIR=FleetCI-RootCA/
+
+export HELM_PATH=/usr/bin/helm
```

Comment:
I would set those variables only if they are not set already, which allows customizing their values (e.g. `HELM_PATH` on Mac is under the Homebrew installation directory):

```
export HELM_PATH="${HELM_PATH:-/usr/bin/helm}"
```

EDIT: I've just seen that this was probably the purpose of the `CUSTOM_CONFIG_FILE`, so maybe we can get rid of that and simplify the scripts a little bit (`setup-cluster-config` no longer needed?)

Reply:
Yes, that is the purpose of `setup-cluster-config`. Considering that the `setup-{single,multi}-cluster` scripts are recommended to be sourced ([see this explanation](https://github.com/rancher/fleet/pull/1764#discussion_r1325508492)) to set the environment variables correctly, changing them to not overwrite previously set environment variables would not change them, e.g.

```bash
source dev/setup-single-cluster && ginkgo e2e/single-cluster
```

would set the environment variables for "single-cluster". A successive run of

```bash
./dev/k3d-act-clean && source dev-setup-multi-cluster && ginkgo e2e/multi-cluster
```

would, provide your suggestions was implemented, not update the environment variables from single to multi cluster. But I understand how convenient it would be to just have to `export HELM_PATH=...` globally once. Perhaps `HELM_PATH` is an exception to those environment variables, since it's not supposed to change. But the same could be said for `CI_OCI_` and `GIT_HTTP_` environment variables. Adding your suggestion to all of them may complicate those scripts, since we'd be having two types of environment variables. At this point I think it may be simpler to create a `env.single-cluster` and `env.multi-cluster` file to customize the setup.

But generally, we can

- leave it as it is,
- change it just for the `HELM_PATH` variable or
- change it for most of the variables, but not all of them.

What do you think?

Reply:
Agree on the 2 types of variables:
 - `FLEET_E2E_*` ones are probably not meant to be modified. If that's the case then it means we are under a different test scenario and they'll probably deserve its own script
 -  Git and OCI user/password, certificates directory or Helm path may be dependent on your local setup and are independent from the test scenario, so I think it makes sense to respect if values are already set (e.g. using `direnv`).